proofs.go

```diff
@@ -808,6 +808,14 @@ func fromFilPoStFaultySectors(ptr []uint64, l uint) ([]abi.SectorNumber, error)
 		return nil, nil
 	}
 
+	type sliceHeader struct {
+		Data unsafe.Pointer
+		Len  int
+		Cap  int
+	}
+
+	(*sliceHeader)(unsafe.Pointer(&ptr)).Len = int(l) // don't worry about it
```

Comment:
Is this needed? Isn't just the fact that we are inerating up to `l` below enough?

Reply:
The generated cgo code doesn't set it, si we panic on out-of-bounds